pulsar-broker/src/main/java/org/apache/pulsar/utils/StringUtil.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringUtil {
+    private static String regEx = "[ _`~!@#$%^&*()+=|{}':;',\\[\\].<>/?~！@#￥%……&*（）——+|{}【】‘；：”“’。，、？]|\n|\r|\t";
```

Comment:
Is there some lib that can check if a char is printable or visible?
IMO, '.' should be here.

BTW, does topic name have the same issue?

Reply:
> BTW, does topic name have the same issue?

Makes sense, I'll check

Reply:
> Is there some lib that can check if a char is printable or visible?
IMO, '.' should be here.

OK, I agree to only check for invisible characters

Reply:
I didn't find some lib that can check if a char is printable or visible.
But I found that the decimal number of invisible characters is 0~31, so I added a method org.apache.pulsar.utils.StringUtil#invisibleCharacters to check if the subscription name contains invisible characters.

![image](https://user-images.githubusercontent.com/19296967/189639602-37f514f1-62e7-4ab8-84ff-4e89a4eac860.png)

Reply:
@Jason918 PTAL,thanks!

I didn't find some lib that can check if a char is printable or visible.
The ASCII values of invisible characters are 0~32 and 127, so I added a method org.apache.pulsar.utils.StringUtil#invisibleCharacters to check if the subscription name contains invisible characters.
<img width="864" alt="image" src="https://user-images.githubusercontent.com/19296967/189642167-e73fe0d6-6d11-431f-9754-925dfcb09bfc.png">

Reply:
> Is there some lib that can check if a char is printable or visible? IMO, '.' should be here.
> 
> BTW, does topic name have the same issue?

@Jason918 PTAL,thanks!

I didn't find some lib that can check if a char is printable or visible.
The ASCII values of invisible characters are 0~32 and 127, so I added a method org.apache.pulsar.utils.StringUtil#invisibleCharacters to check if the subscription name contains invisible characters.
<img width="864" alt="image" src="https://user-images.githubusercontent.com/19296967/189642167-e73fe0d6-6d11-431f-9754-925dfcb09bfc.png">

Reply:
> BTW, does topic name have the same issue?

Yes，I added a check for invisible characters in topic name！

PTAL，thanks! @Jason918

Reply:
@lordcheng10 The problem is that Java uses Unicode by default. There are a lot of helper method in `java.lang.Character`, Please take a look at the methods like `isWhitespace(char)`  `getType(char)`. I am not certain if we can cover all the cases easily. It's worth a try.

Reply:
isWhitespace does not cover all invisible characters

Reply:
@Jason918 
fixed! PTAL,thanks!

Reply:
Invisible characters are generally of these types:
                Character.CONTROL:
                Character.FORMAT:
                Character.PRIVATE_USE:
                Character.SURROGATE:
                Character.UNASSIGNED:
                Character.OTHER_SYMBOL:
                Character.SPACE_SEPARATOR: